### DISCUSSION TITLE
Questions about coverage_map

### DISCUSSION THREAD

**Discussion by JackYifan** [NVlabs/sionna#643] [Q&A]
I am very interested in the code about coverage_map. But I can't find the code about the path length. Whether path lengths are considered in the calculation of the coverage map.
![image](https://github.com/user-attachments/assets/1ea0447c-e4f1-4938-a345-1b010ce8f039)
![image](https://github.com/user-attachments/assets/3515a0a1-30bc-463a-96bd-623212af9d5c)
![image](https://github.com/user-attachments/assets/a60edfb7-0117-4c6c-b29c-1253d117ad7f)

**Comment by faycalaa** [comment]
The path length $\frac{1}{r}$ term from free space propagation cancels out when computing the coverage map through Monte Carlo integration ([see eq. 53](https://nvlabs.github.io/sionna/api/rt.html#equation-cm-mc-ref)) due to the term $r^2$. This is implemented by applying neither the free space propagation loss nor the scaling by $r^2$.

You can also look at post #628, which addresses a similar topic.

**Reply by JackYifan** [reply]
@faycalaa What about the term $e^{-jk_{0}r}$

**Reply by faycalaa** [reply]
This term corresponds to a global phase shift of the path coefficient that has no impact on its squared amplitude and is therefore not applied.

**Reply by JackYifan** [reply]
I'm currently trying to implement refraction, in this case would I need to add this term because k is a complex which will lead to decay of the electric field

**Reply by merlinND** [reply]
Just a head's up that we are working on adding refraction: https://github.com/NVlabs/sionna/discussions/528#discussioncomment-10150663

Note also that there is an open pull request with refraction support: https://github.com/NVlabs/sionna/pull/546

**Reply by JackYifan** [reply]
@merlinND I currently have an urgent task that requires me to calculate refractions, so I've been waiting for this feature for a long time. Could you please tell me when this feature will be released?

**Reply by jhoydis** [reply]
Hi @JackYifan,

Unfortunately, we cannot comment on release dates for new features. Hope for your understanding.

**Reply by potatoess11** [reply]
> @merlinND I currently have an urgent task that requires me to calculate refractions, so I've been waiting for this feature for a long time. Could you please tell me when this feature will be released?

hi,did you have any good idears to achieve this goal?
